MicroERP.API/Controllers/AuthController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using MicroERP.API.Data;
+using MicroERP.ModelsDB.Models.MasterData.Users;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using System.Diagnostics;
+using System.Security.Cryptography;
+using MicroERP.API.Models;
+
+namespace MicroERP.API.Controllers
+{
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly DataContext _context;
+        private readonly IConfiguration _configuration;
+
+        public AuthController(DataContext context, IConfiguration configuration)
+        {
+            _context = context;
+            _configuration = configuration;
+        }
+
+        [HttpPost("Login")]
+        public async Task<ActionResult<User>> Login(Login userLogin)
+        {
+            bool userExists = _context.User.Any(e => e.Name == userLogin.Name) ? true : _context.User.Any(e => e.Email == userLogin.Name);
```

Comment:
Acredito que a verificação do usuário pode ser simplificada. Em vez de fazer duas consultas separadas no banco de dados, seria legal ter apenas uma consulta que verifica a existência do usuário.

```suggestion
var user = _context.User.FirstOrDefault(e => e.Name == userLogin.Name || e.Email == userLogin.Name);

if (user == null)
    return BadRequest("Usuário não existe");

if (VerifyPasswordHash(userLogin.Password, user.Password))
    return Ok(CreateToken(user));
```